src/librustdoc/passes/collect_intra_doc_links.rs

```diff
@@ -59,12 +59,7 @@ fn filter_assoc_items_by_name_and_namespace(
     ident: Ident,
     ns: Namespace,
 ) -> impl Iterator<Item = &ty::AssocItem> {
-    let iter: Box<dyn Iterator<Item = &ty::AssocItem>> = if !ident.name.is_empty() {
-        Box::new(tcx.associated_items(assoc_items_of).filter_by_name_unhygienic(ident.name))
-    } else {
-        Box::new([].iter())
-    };
-    iter.filter(move |item| {
+    tcx.associated_items(assoc_items_of).filter_by_name_unhygienic(ident.name).filter(move |item| {
```

Comment:
Unless I missed something, https://github.com/rust-lang/rust/blame/master/compiler/rustc_middle/src/ty/assoc.rs#L255 is still failing if the provided `Symbol` is empty. How come it's not triggering the `assert` anymore?

Reply:
As per the commit message, "he subsequent changes in https://github.com/rust-lang/rust/pull/140252 means it
is no longer necessary. (Indeed, `Ident`s cannot be empty any more.)"  More specifically, [this part](https://github.com/rust-lang/rust/pull/140252/files#diff-53e2bb31a1a4804925f22caf5a4844d0b6ffc9f7f307d93ef6098d62381a5085) means the empty string case is handled earlier.

Reply:
Should this assert be removed as well then?

Reply:
(Can be done in a follow-up though)

Reply:
The assertion in `filter_by_name_unhygienic`? The function takes a `Symbol`, which can be empty (unlike `Ident`) so I figure it was reasonable to leave it there.

Reply:
Oh I see. A bit confusing. :smile: